tests/k8s/Vagrantfile

```diff
@@ -79,6 +87,28 @@ Vagrant.configure(2) do |config|
         s.inline = "sudo sed -i '/tty/!s/mesg n/tty -s \\&\\& mesg n/' /root/.profile"
     end
 
+    # Set up k8s0 as Docker registry, running newer version of Docker to support staged
+    # builds and not running k8s.  This VM will be idle after k8s[12] have pulled the
+    # newly built cilium image.
+    config.vm.define "k8s0#{$build_id_name}" do |s|
+        s.vm.hostname = "k8s-0"
+
+        s.vm.provider "virtualbox" do |vb|
+            # Do not inherit DNS server from host, use proxy
+            vb.customize ["modifyvm", :id, "--natdnshostresolver1", "on"]
+            vb.customize ["modifyvm", :id, "--natdnsproxy1", "on"]
+            vb.memory = 3072
```

Comment:
Do we need this amount of memory and cpu for the registry?

Reply:
It is not just registry, but also the build VM. On my testing C++ build of Cilium Envoy filters fails with 2GB, but succeeds with 3GB. Also, we delete this VM before we we start running the tests in the other two, so the peak memory use should not increase much.

Reply:
An enhancement as a follow-up could be to use a lighter-weight image - doesn't have to be a part of this ticket, though IMO.

Reply:
renamed the new VM as Ray suggested. It is now called 'builder' which should avoid future confusion here as well :-)

Reply:
I tested builder with 1 CPU and 1GB of memory and it works, so I'll modify this PR accordingly. I'll also roll back the change in number of CPUs on the k8s nodes, I wanted to test if it impacts the runtime but then did not follow through actually timing it.